### PR TITLE
Add Objectio's ReadAllMeta interface

### PR DESCRIPTION
**Pull request by LeftHandCold**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:
Add Objectio's ReadAllMeta interface, supports reading the metadata of all blocks in an object, the checkpoint in tae needs to be used

### CODE REVIEW

File: pkg/objectio/object.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ package objectio
 
 import "github.com/matrixorigin/matrixone/pkg/fileservice"
 
-const Magic = 0xFFFFFFFF
+const Magic = 0xFFFFFFFFFFFFFFFF
```
Comment (fengttt) [inline pkg/objectio/object.go:19]: This is not a good magic value.   It clashes with -1, too often, some code will produce a -1.
Use a strange one ...

**Review by fengttt — APPROVED** [review]
Pull in CN folks please.   Thank you.

**Review by fengttt — CHANGES_REQUESTED** [review]
At least get a few CN people to look at this.

File: pkg/objectio/object.go (hunk 1 above)
Comment (LeftHandCold) [reply pkg/objectio/object.go]: removed

**Review by ouyuanning — APPROVED** [review]

**Review by reusee — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Comment by nnsgmsone** [comment]
Since the interface is not for cn, I'll just approve it.

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**